include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -6590,6 +6628,99 @@ struct OrtApi {
+   * non-ONNX Runtime workloads, such as graphics pipelines (e.g., Direct3D 12 or Vulkan).
+   *
+   * The external synchronization primitive (such as a fence or semaphore handle) referenced by `extSemFence` should be
+   * obtained from a previous call to GetOrtFenceForGraphicsInterop.
+   *
+   * The implementation and support for this is execution provider-specific.
+   *
+   * \param[in] session The OrtSession instance for which synchronization is required.
+   * \param[in] extSemFence The handle to the external synchronization primitive, as returned from GetOrtFenceForGraphicsInterop.
+   * \param[in] stream The OrtSyncStream instance on which the synchronization will be performed.
+   * \param[in] fenceValue The fence value for synchronization (if required for the specific graphics API/interop scenario).
+   *
+   * \retval ORT_OK On successful synchronization.
+   * \retval ORT_NOT_IMPLEMENTED If the current execution provider does not support graphics interop wait.
+   * \retval ORT_FAIL or provider-specific error if the synchronization operation fails for another reason.
+   *
+   * \since Version 1.24
+   */
+  ORT_API2_STATUS(InteropEpWait, _In_ OrtSession* session, _In_ OrtFence* ortFence, _In_ OrtSyncStream* stream, _In_ uint64_t fenceValue);
+
+  /**
+   * \brief Signal a graphics interop external fence/semaphore using an ONNX Runtime execution provider.
+   *
+   * This function synchronizes external graphics or compute APIs with ONNX Runtime computation by signaling an external
+   * semaphore or fence. This is typically used when GPU computation in ONNX Runtime has completed and an external
+   * workload (such as a graphics pipeline) should continue execution.
+   *
+   * The external synchronization primitive (such as a fence or semaphore handle) referenced by `extSemFence` should
+   * have been obtained from a previous call to GetOrtFenceForGraphicsInterop.
+   *
+   * The behavior and support for this operation is execution provider-specific.
+   *
+   * \param[in] session   The OrtSession instance corresponding to the computation that should be synchronized.
+   * \param[in] extSemFence The handle to the external synchronization primitive, as returned from GetOrtFenceForGraphicsInterop.
+   * \param[in] stream    The OrtSyncStream on which the signal operation should occur.
+   * \param[in] fenceValue The fence value to signal (if required for the specific graphics API or interop scenario).
+   *
+   * \retval ORT_OK                On successful signal operation.
+   * \retval ORT_NOT_IMPLEMENTED   If the current execution provider does not support graphics interop signaling.
+   * \retval ORT_FAIL or provider-specific error if signaling fails for another reason.
+   *
+   * \since Version 1.24
+   */
+  ORT_API2_STATUS(InteropEpSignal, _In_ OrtSession* session, _In_ OrtFence* ortFence, _In_ OrtSyncStream* stream, _In_ uint64_t fenceValue);
```

Comment:
Not sure about the naming as the usage of EP/s within a session is somewhat opaque. 

I would have expected the wait/signal to be around a call to OrtSession Run, with ORT figuring out when/where an EP is connected to a fence/semaphore for input/output. The current setup also means you could not have concurrent calls to Run.